### PR TITLE
DHFPROD-6413: Provenance records persisted for each step in a flow with unique id

### PR DESCRIPTION
**Pull request by rahulvudutala**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by ryanjdew — APPROVED** [review]

**Review by srinathgit — DISMISSED** [review]

**Review by Sanjeevani19 — COMMENTED** [review]
@rahulvudutala Few core unit tests are failing on this PR https://jenkins.marklogic.com/blue/organizations/jenkins/Datahub_CI/detail/PR-5364/6/pipeline, Please look into them

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs [hunk 1]
```diff
@@ -241,7 +241,7 @@ class Provenance {
    *      - location (doc URI)
    */
   _createIngestionStepRecord(jobId, flowId, stepName, stepDefinitionName, docURI, info) {
-    let provId = `${jobId + flowId + 'ingestion' + docURI}`;
+    let provId = `${jobId + flowId + 'ingestion' + docURI}_${sem.uuidString()}`;
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs:244]: Per the write-up in 6413, I think all we need is a UUID. The jobId itself is a UUID, so we now have two UUIDs in this. Also see the "How provenance works" Wiki page for details on how this is constructed. 

There's also a lot of duplicated code in this library; we don't want to continue that pattern, we want to clean that up. 

So add a "newProvId()" function that each of these functions uses, and that can just return a UUID.

File: marklogic-data-hub/src/test/java/com/marklogic/hub/flow/ProvenanceDocumentsForEachStepTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ProvenanceDocumentsForEachStepTest extends AbstractHubCoreTest {
+    @Test
+    void verifyProvenanceDocumentsAreGeneratedForEachStep() {
+        installProjectInFolder("test-projects/addSources-step-definition-test");
+        FlowInputs inputs = new FlowInputs("addSourcesFlow", "1", "2");
+        addAbsoluteInputFilePath(inputs, "data");
+        runFlow(inputs);
+        assertEquals(2, getDocCount(HubConfig.DEFAULT_JOB_NAME, "http://marklogic.com/provenance-services/record"));
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/java/com/marklogic/hub/flow/ProvenanceDocumentsForEachStepTest.java:16]: This really needs an assertion message to explain what's going on - i.e. why is 2 expected?

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs (hunk 1 above)
Comment (rahulvudutala) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs:244]: The provenanceId being a UUID will be addressed in a separate ticket

Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/prov.sjs:244]: Let's still get a "newProvId(jobId, flowId, stepType, docUri)" function in place - that avoids duplicating the logic of how the prov ID is constructed in many different places.

**Review by rjrudin — CHANGES_REQUESTED** [review]

**Review by ryanjdew — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Review by rjrudin — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**